docs/DeveloperGuide.md

```diff
@@ -149,6 +149,119 @@ Classes used by multiple components are in the `seedu.storemando.commons` packag
 
 This section describes some noteworthy details on how certain features are implemented.
 
+
+### Add feature 
+
+The `Add` feature allows users to add an item to the inventory. An item is made up of a name, quantity, location, 
+expiry date, and tags.
+
+<div markdown="span" class="alert alert-info">
+:information_source: **Note:** 
+An item's name, quantity and location are compulsory fields that must be supplied by the user.
+</div>
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user
+issues the command `add n/apple q/2 l/kitchen`.
+![AddSequenceDiagram][images/AddSequenceDiagram.png]
+
+####Implementation
+From the diagram above:
+
+1. `LogicManager`'s `execute` is called when the user enters an input into the command box. 
+2. `LogicManager` then calls `parseCommand` of `StoreMandoParser` to parse the user input.
+3.  The `StoreMandoParser` identifies the input as an `AddCommand` and initializes `AddCommandParser`. 
+4. `StoreMando` then invokes the method `parse` of `AddCommandParser` to further parse the user input. 
+   The `AddCommandParser` ensures that the input is of the correct format and identifies the input for name, quantity, 
+   location, expiry date and tag(s).
+5. If all the arguments of the `add` command are valid, The `AddCommandParser` creates a new `Item` object, 
+   and initiates a new `AddCommand` object that contains the `Item` object. This `AddCommand` object will be
+   returned to the `LogicManager`.
+6. The `LogicManager` will then invoke the `execute` method of the `AddCommand` object.   
+7. The `AddCommand` object will then add the new item to `Model`, and return a `CommandResult` to `LogicManager`.
```

Comment:
maybe can consider reduct the usage of 'then' and 'will then'

Reply:
Done!